src/lib/ingest.js

```diff
@@ -1,51 +1,148 @@
+    }
+    return bulkOperations
+  }, [])
+  return operations
+}
+
+export async function writeRecordToDb(
+  /** @type {{ index: string; id: string; body: {}; }} */ record
+) {
+  const { index, id, body } = record
+  const client = await dbClient()
+
+  // is this needed or will update just fail anyway and move on?
+  if (index !== COLLECTIONS_INDEX) {
+    // if this isn't a collection check if index exists
+    const exists = await client.indices.exists({ index })
+    if (!exists.body) {
+      const msg = `Index ${index} does not exist, add before ingesting items`
+      logger.debug(msg)
+      throw new Error(msg)
+    }
+  }
+
+  const result = await client.index({
+    index,
+    type: '_doc',
+    id,
+    body
   })
-  readable.push(item)
-  readable.push(null)
-  return promise
-}
-
-export async function ingestItems(items, stream) {
-  const readable = new Readable({ objectMode: true })
-  const { toDB, dbStream } = await stream()
-  const promise = new Promise((resolve, reject) => {
-    pump(
-      readable,
-      toDB,
-      dbStream,
-      (error) => {
-        if (error) {
-          logger.error('Error ingesting', error)
-          reject(error)
-        } else {
-          logger.debug('Ingested item')
-          resolve(true)
-        }
-      }
-    )
+
+  logger.debug(`Wrote document ${id}`)
+
+  // if this was a collection, then add a new index with collection name
+  if (index === COLLECTIONS_INDEX) {
+    await createIndex(id)
+  }
+  return result
+}
+
+export async function writeRecordsInBulkToDb(records) {
+  const body = combineDbObjectsIntoBulkOperations(records)
+  const client = await dbClient()
+  const result = await client.bulk({ body })
+  logger.debug('Result: %j', result)
+  const { errors } = result.body
+  if (errors) {
+    logger.error('Batch write had errors', errors)
+  } else {
+    logger.debug(`Wrote batch of documents size ${body.length / 2}`)
+  }
+}
+
+async function asyncMapInSequence(objects, asyncFn) {
+  const results = []
+  for (const object of objects) {
+    try {
+      // This helper is inteneted to be used with the objects must be processed
+      // in sequence so we intentionally await each iteration.
+      // eslint-disable-next-line no-await-in-loop
+      const result = await asyncFn(object)
+      results.push(result)
+    } catch (error) {
+      results.push(error)
+    }
+  }
+  return results
+}
+
+function logErrorResults(results) {
+  results.forEach((result) => {
+    if (result instanceof Error) {
+      logger.error('Error while ingesting item', result)
+    }
   })
-  items.forEach((item) => readable.push(item))
-  readable.push(null)
-  return promise
+}
+
+export async function ingestItems(items) {
+  const records = await asyncMapInSequence(items, convertIngestObjectToDbObject)
+  const results = await asyncMapInSequence(records, writeRecordToDb)
+  logErrorResults(results)
+  return results
+}
+
+export async function ingestItem(item) {
```

Comment:
I'd be in favor of getting rid of this method altogether, and just having ingestItems. maybe with a vararg param?